articles/141_static_remapping.md

```diff
@@ -0,0 +1,341 @@
+The name is remapped to the right side only if it exactly matches the left side of a rule.
+
+
+## Remapping rule syntax
+This is a proposal for the ROS 2 remapping rule syntax.
+It attempts to be the same as ROS 1 syntax when possible, and different where it supports new features.
+
+Requirements supported by this syntax
+- Remap One Node in a Process
+- Change a Namespace
+- Change a Basename
+- Exact FQN Replacement
+- Exact Relative Name Replacement
+- Remap via Command Line
+- Change the Default Namespace
+
+Requirements not supported
+- Change a Token
+- Pre-FQN Remapping
+
+Special Operators
+- `*` wild card that matches a single token
+- `**` wild card that matches at least one slash and zero or more tokens
+- `:=` divides the two parts of a remapping rule: matching and replacement
+- `~` expands to `/namespace/nodename` and may only be at the very beginning of a match or replacement
+- `__ns` matches the default namespace if used by itself on the match side of a rule
+- `nodename:` prefixed to a rule makes it apply only to a node with that name
+- `\1` - `\9` are replaced with the matched content of a wild card
+
+The operators `*` and `**` are chosen to match globbing behavior in bash.
+`**` behaves similarly as it does in bash>=4.0 with the globstar option set.
+The syntax for `\[1-9]` was taken from the syntax for backreferences in POSIX BRE.
+The operators `~`, `:=`, and `__ns` have the same meaning in ROS 1.
+
+### Applications of the syntax
+The following sections explain how the syntax enables the use cases above.
+
+#### Supporting: Remap One Node in a Process
+Remapping a node in a process requires a way to uniquely identify a node.
+Assuming the node's name is unique in a process, a rule can be prefix with the name and a `:`.
+If the node name is not prefixed, the rule should be applied to all nodes in the process.
+
+*Example*
+
+- Multiple nodes in a process use the name `scan`
+- A rule is given to one node `node1:scan:=scan_filtered`
+- Only node1 uses the rule
+
+
+#### Supporting: Change a Namespace
+There are two cases: changing part of a namespace, and changing the entire namespace.
+The first case requires a wildcard to match the rest of a namespace.
+The second requires a wildcard to match the basename at the end.
+
+*Example: Partial namespace replacement*
+
+- Node uses names `/foo`, `/foo/bar/baz`
+- Node given rule `/foo**:=/fizz\1`
+- Resulting names `/foo`, `/fizz/bar/baz`
+
+*Example: Full namespace replacement*
+
+- Node uses names `/foo/bar/baz`, `/foo/bar/fee/biz`
+- Node given rule `/foo/bar/*:=/bar/foo/\1`
+- Resulting names `/bar/foo/baz`, `/foo/bar/fee/biz`
+
+#### Supporting: Change a Basename
+Changing a basename requires a wildcard which matches the entire namespace.
+The wildcard `**` is useful because it matches every possible namespace.
+
+*Example:*
+
+- Node uses names `/foo`, `/buz/foo`, `/biz/buz/foo`
+- Node given rule `**foo:=\1bar`
```

Comment:
Same here. For readability I would expect the rule to look like this instead: `**/foo` Especially since the definition doesn't mention anything about a trailing slash.